File: main/java/mod/iceandshadow3/compat/CRefWorld.java
package mod.iceandshadow3.compat;

import net.minecraft.world.World;

public class CRefWorld extends BCRefWorld {
	//WARNING: Most functionality should NOT go here, but in BCRefWorld.
	final World world;
	CRefWorld(World w) {
		world = w;
	}
	@Override
	protected World getWorld() {
		return world;
	}

}

File: main/scala/mod/iceandshadow3/compat/world/CRefWorld.scala
package mod.iceandshadow3.compat.world

import net.minecraft.world.World

class CRefWorld(private[compat] val world: World) extends TCRefWorld {
	override def getWorld(): World = world
	//WARNING: This stubbiness is deliberate.
	//Most functionality should NOT go here, but in TCRefWorld.
}
